app/js/trouble.js

```diff
@@ -56,56 +54,52 @@
 				for (i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+      
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this does remove the "making" of a function from the loop, what you want to do here is declare the function outside the loop, or assign a function expression to a variable outside the loop, then pass in the function or variable name as the first argument to the setTimeout call. By invoking the function here, the function will be called as soon as the code executes rather than waiting until the timeout even has fired, which is the intention of the setTimeout call.